gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementStateStore.java

```diff
@@ -32,44 +33,49 @@
  * and allows add/delete and other functions.
  */
 public interface DagManagementStateStore {
-  void addDag(Dag<JobExecutionPlan> dag);
   /**
    * Persist the {@link Dag} to the backing store.
-   * This is not an actual checkpoint but more like a Write-ahead log, where uncommitted job will be persisted
-   * and be picked up again when leader transition happens.
+   * It is named this because it is usually called to checkpoint any changes in {@link Dag} or in its {@link Dag.DagNode}s.
    * @param dag The dag submitted to {@link DagManager}
    */
   void writeCheckpoint(Dag<JobExecutionPlan> dag) throws IOException;
```

Comment:
seems like this has been combined w/ `addDag` (which is great!)... but why not keep that name rather than this one?  `addDag` seems clearer and also aligns better w/ `containsDag` and `getDag`.

relatedly could we rename `cleanUp` to `deleteDag`?

Reply:
Using this name, I wanted to emphasize on its use pattern "It is named this because it is usually called to checkpoint any changes in {@link Dag}"